sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHive.scala

```diff
@@ -605,6 +605,8 @@ private[hive] class TestHiveQueryExecution(
     // Proceed with analysis.
     sparkSession.sessionState.analyzer.executeAndCheck(logical, tracker)
   }
+
+  lazy val dataset = Dataset.ofRows(sparkSession, logical)
```

Comment:
If it's only used once in https://github.com/apache/spark/pull/27552/files#diff-6b519cd0a9dc5b0ee9caab7712d252e9R349 , let's just inline it.